com.adobe.Flash-Player-Projector.json

```diff
@@ -0,0 +1,69 @@
+{
+    "app-id" : "com.adobe.Flash-Player-Projector",
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "3.28",
+    "sdk" : "org.gnome.Sdk",
+    "command" : "flashplayer",
+    "separate-locales": false,
+    "tags": ["proprietary"],
+    "finish-args" : [
+        /* X11 + XShm access */
+        "--share=ipc", "--socket=x11",
+        /* Play sounds */
+        "--socket=pulseaudio"
+    ],
+    "modules" : [
+        {
+            "name" : "flashplayer",
+            "buildsystem" : "simple",
+            "build-commands" : [
+                "install -D -t /app/bin/ apply_extra",
+                "install flashplayer /app/bin",
+                "install -Dm644 -t /app/share/icons/ com.adobe.Flash-Player-Projector.svg",
+                "install -Dm644 -t /app/share/applications/ com.adobe.Flash-Player-Projector.desktop",
+                "install -Dm644 -t /app/share/metainfo/ com.adobe.Flash-Player-Projector.appdata.xml"
+            ],
+            "sources" : [
+                {
+                    "type" : "script",
+                    "dest-filename" : "flashplayer",
+                    "commands": [
+                       "exec /app/extra/bin/flashplayer.bin $@"
+                    ]
+                },
+                {
+                   "type" : "script",
+                   "dest-filename": "apply_extra",
+                   "commands": [
+                       "echo Linked from https://www.adobe.com/support/flashplayer/debug_downloads.html",
+                       "tar xvzf flash_player_sa_linux.x86_64.tar.gz",
+                       "rm -f flash_player_sa_linux.x86_64.tar.gz",
+                       "install -Dm755 flashplayer /app/extra/bin/flashplayer.bin"
```

Comment:
This is duplicated and doesn't belong there.

Reply:
It's not duplicated, and belongs here. This is the original flashplayer binary from the tarball, it's not installed anywhere else. The flashplayer file in /app/bin is a wrapper shell script, as /app/bin isn't writeable when installing the package, and a binary is necessary for flatpak-builder to finish.

Reply:
But why that `install`? You could just change the script command to `exec /app/extra/flashplayer "$@"` (quoting!).

Reply:
Sure. Except that I didn't know extra_data ended up in `/app/extra`. The Spotify package did something similar.